modules/services/MapRouletteService.js

```diff
@@ -0,0 +1,384 @@
+import { select as d3_select } from 'd3-selection';
+import { Extent, Tiler, vecAdd } from '@rapid-sdk/math';
+import RBush from 'rbush';
+
+import { AbstractSystem } from '../core/AbstractSystem';
+import { Task as MapRouletteTask } from '../maproulette/Task';
+import { utilFetchResponse } from '../util';
+import { marked } from 'marked';
+
+const TILEZOOM = 14;
+const MAPROULETTE_API = 'https://maproulette.org/api/v2';
+
+
+/**
+ * `MapRouletteService`
+
+ * Events available:
+ *   'loadedData'
+ */
+export class MapRouletteService extends AbstractSystem {
+
+  /**
+   * @constructor
+   * @param  `context`  Global shared application context
+   */
+  constructor(context) {
+    super(context);
+    this.id = 'maproulette';
+    this.autoStart = false;
+
+    this._cache = null;   // cache gets replaced on init/reset
+    this._deletedChallenges = {};
+    this._tiler = new Tiler().zoomRange(TILEZOOM).skipNullIsland(true);
+  }
+
+
+  /**
+   * initAsync
+   * Called after all core objects have been constructed.
+   * @return {Promise} Promise resolved when this component has completed initialization
+   */
+  initAsync() {
+    return this.resetAsync();
+  }
+
+
+  /**
+   * startAsync
+   * Called after all core objects have been initialized.
+   * @return {Promise} Promise resolved when this component has completed startup
+   */
+  startAsync() {
+      this._started = true;
+  }
+
+
+  /**
+   * resetAsync
+   * Called after completing an edit session to reset any internal state
+   * @return {Promise} Promise resolved when this component has completed resetting
+   */
+  resetAsync() {
+    if (this._cache) {
+      Object.values(this._cache.inflightTile).forEach(controller => this._abortRequest(controller));
+    }
+    this._cache = {
+      tasks: new Map(),    // Map (taskID -> Task)
+      loadedTile: {},
+      inflightTile: {},
+      inflightPost: {},
+      closed: {},
+      comment: {},
+      rtree: new RBush()
+    };
+    this._deletedChallenges = {};
+    return Promise.resolve();
+  }
+
+
+  /**
+   * getData
+   * Get already loaded data that appears in the current map view
+   * @return  {Array}  Array of data
+   */
+  getData() {
+    const extent = this.context.systems.map.extent();
+    let challengeId = d3_select('.challenge-id-input').property('value');
```

Comment:
Very minor issue -  
In "Service code" like this, I try to avoid adding "UI code" like d3-selecting a value out of a field.  Doing this couples the code pretty tightly - if someone changes the `.challenge-id-input` field elsewhere in Rapid, stuff in this part of the code will stop working, and that's unintuitive.

I think a cleaner way to do this is to have the service code keep track of the current `challengeID` in a class property. The UI code elsewhere can get or set this value, but really the service "owns" that bit of state.

I think of it like: the `MapRouletteService` code's job is only to talk to MapRoulette and keep track of all the data retrieved from there - ideally it shouldn't know anything about `input` fields.  Actually the `MapRouletteService` code should even be able to work in an environment with no UI at all, like a CLI tool.

This is tough too because we do have UI code mixed in with Service code in other places - so you might even see it done that way in old code that we haven't cleaned up yet. We had a lot of d3 code in places like `StreetviewService` and `MapillaryService`, for doing the highlighting and styling of markers before we switched to WebGL.  We also have some code there for dealing with the Mapillary viewer, and for this case we really need the Service and the UI to be working together.